tmt/steps/provision/local.py

```diff
@@ -133,6 +133,43 @@ def pull(
             extend_options: Optional[list[str]] = None) -> None:
         """ Nothing to be done to pull workdir """
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """fetch and return content of a requested log"""
+        if log_name == 'dmesg':
+            return self.execute(Command('dmesg')).stdout
+        return None
+
+    def store_log(
+            self,
+            log_name: Optional[str],
+            log_path: Optional[Path],
+            log_content: str) -> None:
+        """save log content and return the path"""
+        if log_path and log_name:
+            # if log_path contains log file name
+            if log_path.name == log_name:
+                log_path.write_text(log_content)
+            else:
+                (log_path / log_name).write_text(log_content)
+        elif log_path and not log_name:
+            log_path.write_text(log_content)
+        elif log_name and not log_path:
+            self.info('no log_path provided,store log to current dir')
+            (Path.cwd() / log_name).write_text(log_content)
+        else:
+            self.warn('log_name and log_path cannot both be None')
+
+    def handle_guest_logs(self, logs: Optional[list[str]] = None) -> None:
+        """get log content and save it to the workdir/provision/,
+        list the log dir in guests.yaml"""
+        logs = logs or []
+        for log_name in logs:
+            log_content = self.acquire_log(log_name)
+            if log_content:
+                self.store_log(log_name, self.workdir, log_content)
+            else:
+                self.debug(f'no content in {log_name}')
```

Comment:
Why not make this the default implementation of `store_log()` and `handle_guest_logs()` in the `Guest` class? There is *nothing* that would be `local`-specific in these two methods, and they can provide a handy default behavior for all modules.

Reply:
there is nothing local-specific in store_log,but what logs can be provided depends on different guests,I was thinking define a LOG_LIST for each plugin,so I put handle_guest_logs to local.py,but when I saw your comment,I suddenly realized that I could define a log_list property for each plugin,and put both of that two methods to Guest class,right?^^

Reply:
Yep, "the list of logs" and "saving of the list of logs" are two distinct parts, and the plugin may easily override the former without changing a bit in the latter.